### ISSUE TITLE
[Stealth] Detected by Google

### ISSUE THREAD

**Issue by userlip** [berstend/puppeteer-extra#193]
Hey,

Since a few days now more and more often my puppeteer instances get detected by Google.
Trying to sign in no longer works.

After putting in my email it says that the browser is not safe to use.

**Comment by nitishmakhija** [comment]
Facing the same issue.
Any update on this?

**Comment by zhzehong** [comment]
+1

**Comment by thegeekybaniya** [comment]
Yes, this has started in the last few days, I had written and tested a script for signing up on Gmail 15 days ago.

**Comment by JunkTurkey** [comment]
+1

**Comment by sandou78** [comment]
I downgrade my plug in to 2.4.5 and is working now

**Comment by JunkTurkey** [comment]
downgrading really works

**Comment by nitishmakhija** [comment]
> I downgrade my plug in to 2.4.5 and is working now

Can you help me I tried downgrading it to 2.4.5 the stealth plugin.

What would be the version of extra and puppeteer dependency?

**Comment by sandou78** [comment]
all you need to do it npm i puppeteer-extra-plugin-stealth@2.4.5

**Comment by JediBurrell** [comment]
Downgrading doesn't work for me.

puppeteer@2.1.0  
puppeteer-extra@3.1.9  
puppeteer-extra-plugin-stealth@2.4.5

```js
const browser = await puppeteer.launch({headless: false});
const page = (await browser.pages())[0];
page.goto('https://accounts.google.com/SignOutOptions?continue=https%3A%2F%2Fstadia.google.com%2F');
```

**Comment by nitishmakhija** [comment]
> all you need to do it  is    -->   npm i puppeteer-extra-plugin-stealth@2.4.5 

Tried it didn’t work 
Can you share a sample package.json file with me ? Showing the versions for dependencies that would be a big help

**Comment by dench1k** [comment]
> Tried it didn’t work
> Can you share a sample package.json file with me ? Showing the versions for dependencies that would be a big help
`{
  "dependencies": {
    "puppeteer": "^2.1.1",
    "puppeteer-extra": "^3.1.9",
    "puppeteer-extra-plugin-stealth": "^2.4.5"
  }
}`

**Comment by nitishmakhija** [comment]
> > Tried it didn’t work
> > Can you share a sample package.json file with me ? Showing the versions for dependencies that would be a big help
> 
> `{ "dependencies": { "puppeteer": "^2.1.1", "puppeteer-extra": "^3.1.9", "puppeteer-extra-plugin-stealth": "^2.4.5" } }`

Nope still getting the same error

**Comment by dench1k** [comment]
> Downgrading doesn't work for me.
> Nope still getting the same error

I can only suggest setting the same options to puppeteer as me:
`browser = await puppeteer.launch({
      args: [
        "--no-sandbox",
        "--headless",
        "--disable-gpu",
        "--window-size=1920x1080",
      ],
    });`

**Comment by Lastofthefirst** [comment]
In my case I am getting detected using gmail, when i automate log in from one account (which i previously automated log in many times), but not from an account which I havnt previously used.

**Comment by itsdarrylnorris** [comment]
Downgrading to 2.4.5 seems to fix this issue.

**Comment by AdityaAnand1** [comment]
Any updates on an actual solution instead of just downgrading?

**Comment by teewoeijer** [comment]
Add this line if you don’t want to don’t want to downgrade.
```

await page.evaluateOnNewDocument(() => {
	const newProto = navigator.__proto__
	delete newProto.webdriver
	navigator.__proto__ = newProto
});

```

**Comment by pashkasan** [comment]
> Add this line if you don’t want to downgrade.
> 
> ```
> 
> await page.evaluateOnNewDocument(() => {
> 	const newProto = navigator.__proto__
> 	delete newProto.webdriver
> 	navigator.__proto__ = newProto
> });
> ```

confirm this works, thanks man!

**Comment by AdityaAnand1** [comment]
I'm very curious to know how come so many people are scraping google, what are your use-cases? How are you using the scraped data?

**Comment by dench1k** [comment]
I make music mixes every month and post it on a couple of platforms including youtube. There is a thing called music policies, where you can search for a track and get the info about it's license (i.e. can you actually use it without restrictions or it can cause ban). So I made a bot, which logs in the gmail account and searches via this service through my tracklist, tells me the info about a track and makes a screenshot to check this info.

**Comment by itsdarrylnorris** [comment]
@AdityaAnand1 ,

I was doing a proof-of-concept on automating login to Gmail, and Google will notice that you are using puppeteer if you are not using this package.
